EG-3165: updating release notes for 4.5.1 (#189)
corda/corda-docs: content/en/docs/corda-enterprise/4.5/release-notes-enterprise.md

# Corda Enterprise release notes

## Corda Enterprise 4.5.1

> Corda Enterprise 4.5.1 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.5.

Reviewer: Please add the upgrade recommendation here too. Although this IS the latest version now, it won't be once we release 4.6 and the upgrade boilerplate points to the actual current latest version, so I think it's good to have it in all patch release notes.

### Upgrade recommendation

As a developer, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) as soon as possible. Check the latest Corda Enterprise release notes and upgrade guide [here](https://docs.corda.net/docs/corda-enterprise/release-notes-index.html).

As a node operator, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) if the fixed issues listed below are relevant to your work.

Author: Done.

> ### Issues Fixed

Reviewer 2: some of the issues are almost described as solutions. This is bit picky, but if you wanted to, you could change 'Command line state ref usability improvement' to something else? Or introduce the list as 

### Issues and solutions

Author: I think this is a bit picky, especially at the patch-release level. It is worth bearing in mind for future major releases though, and we should probably think about what we want to see in all release notes (including patch-level release notes) in Corda 5.

> ### Issues Fixed

Reviewer: ```suggestion
### Fixed issues
```

Reviewer: for consistency with CENM patch RNs etc. - although in future I would rather use "Resolved issues" instead, but maybe from Corda 5 onwards once we can cut the history line:)

> * Classloader fails to find class when Optional generic used on Type definition.

Reviewer 3: Comments left on https://github.com/corda/corda-docs/pull/188#pullrequestreview-465350740 apply here too:

* Consistency of release note entries
* Entries containing square brackets

Reviewer: Same - please see https://github.com/corda/corda-docs/pull/188 and suggestions (for format/style) in https://github.com/corda/corda-docs/pull/190.

* Config Obfuscator doesn't work anymore for HSM configuration files
* Retrying session inits can fail due to database connectivity

> * Revert H2 version to 197 to avoid bug introduced in the dependency after

Reviewer 2: Sentence seems to be incomplete -- after...

Author: I'll follow this up with @fowlerrr - this line came directly from one of his JIRA tickets.

Author: Spoke to @fowlerrr on Slack and added "after previous upgrade." to the end of that bullet point.

> * Revert H2 version to 197 to avoid bug introduced in the dependency after previous upgrade.

Reviewer 3: Actual version is 1.4.197, per `build.gradle` file

Author: Fixed

> * Classloader fails to find class when Optional generic used on Type definition
> * Config Obfuscator doesn't work anymore for HSM configuration files
> * Retrying session inits can fail due to database connectivity
> * Revert H2 version to 197 to avoid bug introduced in the dependency after
> * NodeMeteringBackground processes high CPU while a node is idle
> * Prevent AMQP header spoofing 
> * Passing two sessions with the same counterparty to the CollectSignaturesFlow leads both counterparty's flows to infinitely await messages from other party
> * Uncaught exception in FlowStateMachineImpl.run().initialiseFlow()
> * [ConfigObfuscator] Firewall doesn't start if its main config and HSM config was obfuscated with specified SEED and PASSPHRASE
> * TLS handshake timeout leads to blacklisting endpoint
> * Command line state ref usability improvement (CE 4.5.1)

Reviewer 2: ```suggestion
* Classloader fails to find class when Optional generic used on Type definition.
* Config Obfuscator doesn't work anymore for HSM configuration files.
* Retrying session inits can fail due to database connectivity.
* Revert H2 version to 197 to avoid bug introduced in the dependency after
* NodeMeteringBackground processes high CPU while a node is idle.
* Prevent AMQP header spoofing.
* Passing two sessions with the same counterparty to the CollectSignaturesFlow leads both counterparty's flows to infinitely await messages from other party.
* Uncaught exception in FlowStateMachineImpl.run().initialiseFlow().
* [ConfigObfuscator] Firewall doesn't start if its main config and HSM config was obfuscated with specified SEED and PASSPHRASE.
* TLS handshake timeout leads to blacklisting endpoint.
* Command line state ref usability improvement (CE 4.5.1).
```

> * Command line state ref usability improvement (CE 4.5.1).

Reviewer 3: Would rephrase the last point for clarity to:

"Added support to the spent state audit command for specifying state references in the form `txId(outputIdx)` in addition to the existing `txId:outputIdx`"

(from the ticket description)

Author: Done

# Corda Enterprise release notes

## Corda Enterprise 4.5.1

Corda Enterprise 4.5.1 is a patch release of Corda Enterprise that introduces fixes to known issues in Corda Enterprise 4.5.

### Upgrade recommendation

As a developer, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) as soon as possible. Check the latest Corda Enterprise release notes and upgrade guide [here](https://docs.corda.net/docs/corda-enterprise/release-notes-index.html).

As a node operator, you should upgrade to the [latest released version of Corda](https://docs.corda.net/docs/corda-enterprise/index.html) if the fixed issues listed below are relevant to your work.

### Fixed issues

* Fixed an issue where the Classloader failed to find a Command class when Optional generic was used on Type definition.
* The [Configuraton Obfuscator tool](tools-config-obfuscator.md) has been fixed to work for HSM configuration files.

> * Retrying session inits can fail due to database connectivity.

Reviewer: ```suggestion
* Fixed an issue where retrying session inits could fail due to database connectivity.
```

* The H2 version has been reverted to 1.4.197 to avoid a dependency issue introduced after the previous upgrade.
* The CPU usage of the `NodeMeteringBackground` process has been decreased.
* A security update to prevent AMQP header spoofing has been applied.

> * Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` lead to both counterparties' flows to wait infinitely for messages from the other party.

Reviewer: ```suggestion
* Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` lead to both counterparties' flows having to wait infinitely for messages from the other party.
```

Reviewer 4: lead to both counterparties' flows having to wait infinitely for messages from the other party.

Needs to be "leads" or "led" above

Reviewer: Yep well-spotted! Fixed in two places.

> * Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` lead to both counterparties' flows having to wait infinitely for messages from the other party.

Reviewer: ```suggestion
* Fixed an issue where passing two sessions with the same counterparty to the `CollectSignaturesFlow` led to both counterparties' flows having to wait infinitely for messages from the other party.
```

* A previously unhandled exception in `FlowStateMachineImpl.run().initialiseFlow()` is now handled correctly.
* Fixed an issue where Corda Firewall did not start if its main configuration and its HSM configuration were obfuscated.

> * Fixed an issue where a TLS handshake timeout lead to blacklisting endpoint.

Reviewer: ```suggestion
* Fixed an issue where a TLS handshake timeout led to blacklisting endpoint.
```
